Fix: wrong follow up question

shuchimehta-juspay opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Improvements**
  * Enhanced voice agent's contextual awareness to provide more relevant suggestions and avoid generic topics.
  * Agent now includes mandatory follow-up questions after every response to guide next steps.
  * Clarified which data breakdown types and features the agent cannot suggest, ensuring more focused assistance.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shuchimehta-juspay committed:
Revert "Remove unnecessary errors that don’t need to be raised"

This reverts commit dbab4ddaa2039bbb5d974e2d8a662c61dbe6d755.

shuchimehta-juspay committed:
Fix: wrong follow up question

swaroopvarma1 commented on app/ai/voice/agents/automatic/prompts/system/tool_scope.py:
```diff
@@ -96,26 +96,36 @@ def get_tool_scope_instructions(shop_id: str | None) -> str:
 
-            CONTEXTUAL RELEVANCE RULE: Suggestions MUST directly relate to what was just discussed. Never suggest random and generic topics.
+            CONTEXTUAL RELEVANCE RULE: Suggestions MUST directly relate to what was just discussed. Never suggest random topics.
+            CONTEXTUAL RELEVANCE RULE: Suggestions MUST directly relate to what was just discussed. Never suggest random AND generic topics.
```
Why two CONTEXTUAL RELEVANCE RULE:?

shuchimehta-juspay replied:
Changed it

swaroopvarma1 commented on app/ai/voice/agents/automatic/prompts/system/tool_scope.py:
```diff
@@ -96,26 +96,36 @@ def get_tool_scope_instructions(shop_id: str | None) -> str:
             - E-commerce Metrics → Conversion rates/address completion/marketing attribution
             - General/Greetings → Business summary/today's performance/key metrics
 
+            UNAVAILABLE FEATURES - DO NOT SUGGEST:
```
Category-wise detailed breakdowns (unless already present in the data shown)

-> This is a negative and Positive prompt both inside a Negative prompt section, which will cause hallucination

swaroopvarma1 commented on app/agents/voice/automatic/prompts/system/tool_scope.py:
```diff
@@ -0,0 +1,135 @@
+from app.core.config.static import (
```
Rebase PR

shuchimehta-juspay committed:
fix: align follow-up suggestions with available tools

- Remove contradictory follow-up rules by adding exceptions for clarification/error/confirmation scenarios
- Deduplicate contextual relevance rule (remove 'random AND generic' ambiguity)
- Separate negative constraints (UNAVAILABLE FEATURES) from positive permissions (ALLOWED ENHANCEMENTS)
- Update Order Metrics pattern to list only available breakdowns
- Make follow-ups conditional: only after data responses, not during interactive flows

Resolves BZ-45053

shuchimehta-juspay committed:
fix: apply same follow-up suggestions fixes to ai/voice path

Apply the same improvements to the duplicate tool_scope.py file:
- Align Response Protocol with conditional follow-ups
- Deduplicate contextual relevance rule
- Separate UNAVAILABLE FEATURES from ALLOWED ENHANCEMENTS
- Make follow-ups conditional with exceptions for interactive flows

Addresses reviewer feedback on duplicate file path

shuchimehta-juspay committed:
revert: remove unrelated changes to hitl.py and configuration.py

swaroopvarma1 reviewed (commented):
Why these many prompt changes?
What is failing now and why is this needed?

shuchimehta-juspay committed:
fix: remove unused duplicate tool_scope.py file

The file app/agents/voice/automatic/prompts/system/tool_scope.py has no imports and is not used anywhere.
Only app/ai/voice/agents/automatic/prompts/system/tool_scope.py is actively imported and used.
Removing the unused duplicate to keep PR focused on fixing the actual active file.

shuchimehta-juspay commented:
> Why these many prompt changes? What is failing now and why is this needed?

Without this:
Follow-up questions suggested are about product level breakdown and customer level breakdown. for which tools are not available now.
